### ISSUE TITLE
False positive with Euro Symbol

### ISSUE THREAD

**Issue by n2o** [valentjn/vscode-ltex#25]
**Describe the bug**
When a sentence is ending with `\euro{}`, LTeX somehow interprets it as a whitespace. 

Sample:

```latex
\usepackage{eurosym}
```

```latex
Let's talk about 10 \euro{}.
```

![image](https://user-images.githubusercontent.com/1507474/74238435-13601e00-4cd6-11ea-93f4-0ced6f81c038.png)

**Expected behavior**
This is no error and should not be linted.

**LTeX configuration**
Default.

**Log files**
First, set the `ltex.trace.server` setting in your `settings.json` to `"verbose"`. Then, reload the VS Code window and reproduce the bug. Go to `View` → `Output` and select `LTeX Language Server` and `LTeX Language Client` in the drop-down list. Paste or attach the relevant parts of these logs here in the GitHub issue.

LTeX Language Server
```log
INFORMATION: Checking the following text in language en-US via LanguageTool: "Let's talk about 10 . "
Feb 11, 2020 2:22:00 PM org.bsplines.languagetool_languageserver.LanguageToolLanguageServer lambda$validateDocument$3
INFORMATION: Obtained 1 rule match
```

LTeX Language Client
```log
[Trace - 2:22:38 PM] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///path/to/my.tex",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 0,
                    "character": 20
                },
                "end": {
                    "line": 0,
                    "character": 28
                }
            },
            "severity": 3,
            "source": "LTeX - COMMA_PARENTHESIS_WHITESPACE",
            "message": "Don't put a space before the full stop"
        }
    ]
}
```

**Version information**
- macOS 10.15.3
- VSCode 1.42.0
- LTeX: 4.77
- LTeX language extensions: English
- Java: 1.8.0_152-release

**Comment by n2o** [comment]
Usually I'd write it more like this:

```latex
10~\euro{}
```

but this results in the same error. Maybe the combination of `~` and a latex command `\euro{}` *together* should get removed instead of the symbol `\euro{}` alone.

**Comment by n2o** [comment]
Thank you!

**Comment by valentjn** [comment]
@n2o No worries, thank you for the detailed bug report :smiley:

As you probably guessed, all unknown L<sup>A</sup>T<sub>E</sub>X commands (like `\euro` was) are removed by default.

`\euro` shouldn't be removed at all, but rather replaced with `€`, as LanguageTool is perfectly capable of dealing with currency symbols. For example, LT warns for German texts if something like `10€` is found, because there should be a space between the number and the currency symbol.

The fix will be in the upcoming release of LT<sub>E</sub>X.

**Comment by valentjn** [comment]
Fixed in 4.7.8.
